File: bucket_1/1.py
import requests
from google.protobuf.json_format import MessageToDict
from google.protobuf import field_mask_pb2

# endpoint URL
BASE_URL = "https://api.aip_research.com/v1"

# Request headers
headers = {
    "Content-Type": "application/json",
    "Authorization": "Bearer ACCESS_TOKEN",
}

class Account:
    def __init__(self, name="", display_name="", birthdate="", email="", password="", username="", isParent=False, bio="", children=None):
        self.name = name
        self.display_name = display_name
        self.birthdate = birthdate
        self.email = email
        self.password = password
        self.username = username
        self.isParent = isParent
        self.bio = bio
        self.children = children or []

class CreateAccountRequest:
    def __init__(self, account=None):
        self.account = account

class VerifyAccountRequest:
    def __init__(self, secret=""):
        self.secret = secret

class UpdateAccountRequest:
    def __init__(self, account=None, update_mask=None):
        self.account = account
        self.update_mask = update_mask

class ListAccountsRequest:
    def __init__(self, page_size=10, page_token="", filter=""):
        self.page_size = page_size
        self.page_token = page_token
        self.filter = filter

class CreateTweetRequest:
    def __init__(self, tweet=None):
        self.tweet = tweet

class DeleteTweetRequest:
    def __init__(self, name=""):
        self.name = name

class ListTweetsRequest:
    def __init__(self, page_size=10, page_token="", filter="", show_deleted=False, order_by=""):
        self.page_size = page_size
        self.page_token = page_token
        self.filter = filter
        self.show_deleted = show_deleted
        self.order_by = order_by

class ExportTweetsRequest:
    def __init__(self, parent="", csv_destination="", filter=""):
        self.parent = parent
        self.csv_destination = csv_destination
        self.filter = filter

class CreateLikeRequest:
    def __init__(self, parent="", like=None):
        self.parent = parent
        self.like = like

class CreateFolloweeRequest:
    def __init__(self, parent="", followee=None):
        self.parent = parent
        self.followee = followee

# API requests functions
def create_account(account):
    url = f"{BASE_URL}/accounts"
    payload = CreateAccountRequest(account=account)
    response = requests.post(url, json=MessageToDict(payload), headers=headers)
    return response

def verify_account(secret):
    url = f"{BASE_URL}/accounts:verify"
    payload = VerifyAccountRequest(secret=secret)
    response = requests.post(url, json=MessageToDict(payload), headers=headers)
    return response

def update_account(account, update_mask):
    url = f"{BASE_URL}/{account.name}"
    payload = UpdateAccountRequest(account=account, update_mask=update_mask)
    response = requests.patch(url, json=MessageToDict(payload), headers=headers)
    return response

def list_accounts(page_size=10, page_token="", filter=""):
    url = f"{BASE_URL}/accounts"
    payload = ListAccountsRequest(page_size=page_size, page_token=page_token, filter=filter)
    response = requests.get(url, params=MessageToDict(payload), headers=headers)
    return response

def create_tweet(tweet):
    url = f"{BASE_URL}/tweets"
    payload = CreateTweetRequest(tweet=tweet)
    response = requests.post(url, json=MessageToDict(payload), headers=headers)
    return response

def delete_tweet(name):
    url = f"{BASE_URL}/{name}"
    payload = DeleteTweetRequest(name=name)
    response = requests.delete(url, json=MessageToDict(payload), headers=headers)
    return response

def list_tweets(page_size=10, page_token="", filter="", show_deleted=False, order_by=""):
    url = f"{BASE_URL}/tweets"
    payload = ListTweetsRequest(page_size=page_size, page_token=page_token, filter=filter, show_deleted=show_deleted, order_by=order_by)
    response = requests.get(url, params=MessageToDict(payload), headers=headers)
    return response

def export_tweets(parent, csv_destination, filter=""):
    url = f"{BASE_URL}/tweets:export"
    payload = ExportTweetsRequest(parent=parent, csv_destination=csv_destination, filter=filter)
    response = requests.post(url, json=MessageToDict(payload), headers=headers)
    return response

def create_like(parent, like):
    url = f"{BASE_URL}/{parent}/likes"
    payload = CreateLikeRequest(parent=parent, like=like)
    response = requests.post(url, json=MessageToDict(payload), headers=headers)
    return response

def create_followee(parent, followee):
    url = f"{BASE_URL}/{parent}/followees"
    payload = CreateFolloweeRequest(parent=parent, followee=followee)
    response = requests.post(url, json=MessageToDict(payload), headers=headers)
    return response

# Usage example
if __name__ == "__main__":
    new_account = Account(
        display_name="Ridwan Abdusalam",
        email="rtabdusalam@ucdavis.edu",
        password="password123",
        isParent=True
    )

    account_response = create_account(new_account)
    if account_response.status_code == 200:
        print("Account created successfully!")
    else:
        print("Error creating account:", account_response.text)

    verification_response = verify_account("your_secret_key_here")
    if verification_response.status_code == 200:
        print("Account verified successfully!")
    else:
        print("Error verifying account:", verification_response.text)

    update_mask = field_mask_pb2.FieldMask(paths=["display_name"])
    updated_account = Account(
        name="accounts/123456",  # Replace with actual account name
        display_name="New Display Name"
    )

    update_response = update_account(updated_account, update_mask)
    if update_response.status_code == 200:
        print("Account updated successfully!")
    else:
        print("Error updating account:", update_response.text)

    list_accounts_response = list_accounts(page_size=10)
    if list_accounts_response.status_code == 200:
        accounts = list_accounts_response.json().get("accounts", [])
        print("List of accounts:", accounts)
    else:
        print("Error listing accounts:", list_accounts_response.text)

    new_tweet = Tweet(
        text="First tweet!",
        author="accounts/123456"  # Replace with actual account name
    )

    tweet_response = create_tweet(new_tweet)
    if tweet_response.status_code == 200:
        print("Tweet created successfully!")
    else:
        print("Error creating tweet:", tweet_response.text)

    delete_tweet_response = delete_tweet("tweets/789012")  # Replace with actual tweet name
    if delete_tweet_response.status_code == 204:
        print("Tweet deleted successfully!")
    else:
        print("Error deleting tweet:", delete_tweet_response.text)

    list_tweets_response = list_tweets(page_size=10)
    if list_tweets_response.status_code == 200:
        tweets = list_tweets_response.json().get("tweets", [])
        print("List of tweets:", tweets)
    else:
        print("Error listing tweets:", list_tweets_response.text)

    export_tweets_response = export_tweets("accounts/123456", "exported_tweets.csv")
    if export_tweets_response.status_code == 200:
        print("Tweets exported successfully!")
    else:
        print("Error exporting tweets:", export_tweets_response.text)

    new_like = Like(
        account="accounts/123456",  # Replace with actual account name
        tweet="tweets/789012"  # Replace with actual tweet name
    )

    like_response = create_like("tweets/789012", new_like)
    if like_response.status_code == 200:
        print("Like created successfully!")
    else:
        print("Error creating like:", like_response.text)

    new_followee = Followee(
        followee="accounts/789012"  # Replace with actual account name
    )

    followee_response = create_followee("accounts/123456", new_followee)
    if followee_response.status_code == 200:
        print("Followee created successfully!")
    else:
        print("Error creating followee:", followee_response.text)

File: bucket_1/1.js
const axios = require('axios');

// Define your API endpoint URL
const BASE_URL = 'https://api.example.com/v1';

// Define the headers for your requests (if needed)
const headers = {
  'Content-Type': 'application/json',
  Authorization: 'Bearer YOUR_ACCESS_TOKEN',
};

class Account {
  constructor({
    name = '',
    display_name = '',
    birthdate = '',
    email = '',
    password = '',
    username = '',
    isParent = false,
    bio = '',
    children = [],
  }) {
    this.name = name;
    this.display_name = display_name;
    this.birthdate = birthdate;
    this.email = email;
    this.password = password;
    this.username = username;
    this.isParent = isParent;
    this.bio = bio;
    this.children = children;
  }
}

class CreateAccountRequest {
  constructor(account = null) {
    this.account = account;
  }
}

class VerifyAccountRequest {
  constructor(secret = '') {
    this.secret = secret;
  }
}

class UpdateAccountRequest {
  constructor(account = null, update_mask = null) {
    this.account = account;
    this.update_mask = update_mask;
  }
}

class ListAccountsRequest {
  constructor(page_size = 10, page_token = '', filter = '') {
    this.page_size = page_size;
    this.page_token = page_token;
    this.filter = filter;
  }
}

class CreateTweetRequest {
  constructor(tweet = null) {
    this.tweet = tweet;
  }
}

class DeleteTweetRequest {
  constructor(name = '') {
    this.name = name;
  }
}

class ListTweetsRequest {
  constructor(page_size = 10, page_token = '', filter = '', show_deleted = false, order_by = '') {
    this.page_size = page_size;
    this.page_token = page_token;
    this.filter = filter;
    this.show_deleted = show_deleted;
    this.order_by = order_by;
  }
}

class ExportTweetsRequest {
  constructor(parent = '', csv_destination = '', filter = '') {
    this.parent = parent;
    this.csv_destination = csv_destination;
    this.filter = filter;
  }
}

class CreateLikeRequest {
  constructor(parent = '', like = null) {
    this.parent = parent;
    this.like = like;
  }
}

class CreateFolloweeRequest {
  constructor(parent = '', followee = null) {
    this.parent = parent;
    this.followee = followee;
  }
}

// Define functions to make API requests
async function createAccount(account) {
  const url = `${BASE_URL}/accounts`;
  const payload = new CreateAccountRequest({ account });
  const response = await axios.post(url, payload, { headers });
  return response;
}

async function verifyAccount(secret) {
  const url = `${BASE_URL}/accounts:verify`;
  const payload = new VerifyAccountRequest({ secret });
  const response = await axios.post(url, payload, { headers });
  return response;
}

async function updateAccount(account, update_mask) {
  const url = `${BASE_URL}/${account.name}`;
  const payload = new UpdateAccountRequest({ account, update_mask });
  const response = await axios.patch(url, payload, { headers });
  return response;
}

async function listAccounts(page_size = 10, page_token = '', filter = '') {
  const url = `${BASE_URL}/accounts`;
  const payload = new ListAccountsRequest({ page_size, page_token, filter });
  const response = await axios.get(url, { params: payload, headers });
  return response;
}

async function createPost(post) {
  const url = `${BASE_URL}/posts`;
  const payload = new CreatePostRequest({ post });
  const response = await axios.post(url, payload, { headers });
  return response;
}

async function deletePost(name) {
  const url = `${BASE_URL}/${name}`;
  const payload = new DeletePostRequest({ name });
  const response = await axios.delete(url, { data: payload, headers });
  return response;
}

async function listTweets(page_size = 10, page_token = '', filter = '', show_deleted = false, order_by = '') {
  const url = `${BASE_URL}/tweets`;
  const payload = new ListTweetsRequest({ page_size, page_token, filter, show_deleted, order_by });
  const response = await axios.get(url, { params: payload, headers });
  return response;
}

async function exportTweets(parent, csv_destination, filter = '') {
  const url = `${BASE_URL}/tweets:export`;
  const payload = new ExportTweetsRequest({ parent, csv_destination, filter });
  const response = await axios.post(url, payload, { headers });
  return response;
}

async function createLike(parent, like) {
  const url = `${BASE_URL}/${parent}/likes`;
  const payload = new CreateLikeRequest({ parent, like });
  const response = await axios.post(url, payload, { headers });
  return response;
}

async function createFollowee(parent, followee) {
  const url = `${BASE_URL}/${parent}/followees`;
  const payload = new CreateFolloweeRequest({ parent, followee });
  const response = await axios.post(url, payload, { headers });
  return response;
}

// Usage example
(async () => {
  const newAccount = new Account({
    display_name: 'John Doe',
    email: 'john@example.com',
    password: 'password123',
    isParent: true,
  });

  try {
    const accountResponse = await createAccount(newAccount);
    if (accountResponse.status === 200) {
      console.log('Account created successfully!');
    } else {
      console.error('Error creating account:', accountResponse.data);
    }
  } catch (error) {
    console.error('Error creating account:', error);
  }

  try {
    const verificationResponse = await verifyAccount('your_secret_key_here');
    if (verificationResponse.status === 200) {
      console.log('Account verified successfully!');
    } else {
      console.error('Error verifying account:', verificationResponse.data);
    }
  } catch (error) {
    console.error('Error verifying account:', error);
  }

  const updateMask = {
    paths: ['display_name'],
  };
  const updatedAccount = new Account({
    name: 'accounts/123456', // Replace with actual account name
    display_name: 'New Display Name',
  });

  try {
    const updateResponse = await updateAccount(updatedAccount, updateMask);
    if (updateResponse.status === 200) {
      console.log('Account updated successfully!');
    } else {
      console.error('Error updating account:', updateResponse.data);
    }
  } catch (error) {
    console.error('Error updating account:', error);
  }

  try {
    const listAccountsResponse = await listAccounts(10);
    if (listAccountsResponse.status === 200) {
      const accounts = listAccountsResponse.data.accounts || [];
      console.log('List of accounts:', accounts);
    } else {
      console.error('Error listing accounts:', listAccountsResponse.data);
    }
  } catch (error) {
    console.error('Error listing accounts:', error);
  }

  const newTweet = {
    text: 'Hello, world!',
    author: 'accounts/123456', // Replace with actual account name
  };

  try {
    const tweetResponse = await createTweet(newTweet);
    if (tweetResponse.status === 200) {
      console.log('Tweet created successfully!');
    } else {
      console.error('Error creating tweet:', tweetResponse.data);
    }
  } catch (error) {
    console.error('Error creating tweet:', error);
  }

  try {
    const deleteTweetResponse = await deleteTweet('tweets/789012'); // Replace with actual tweet name
    if (deleteTweetResponse.status === 204) {
      console.log('Tweet deleted successfully!');
    } else {
      console.error('Error deleting tweet:', deleteTweetResponse.data);
    }
  } catch (error) {
    console.error('Error deleting tweet:', error);
  }

  try {
    const listTweetsResponse = await listTweets(10);
    if (listTweetsResponse.status === 200) {
      const tweets = listTweetsResponse.data.tweets || [];
      console.log('List of tweets:', tweets);
    } else {
      console.error('Error listing tweets:', listTweetsResponse.data);
    }
  } catch (error) {
    console.error('Error listing tweets:', error);
  }

  try {
    const exportTweetsResponse = await exportTweets('accounts/123456', 'exported_tweets.csv');
    if (exportTweetsResponse.status === 200) {
      console.log('Tweets exported successfully!');
    } else {
      console.error('Error exporting tweets:', exportTweetsResponse.data);
    }
  } catch (error) {
    console.error('Error exporting tweets:', error);
  }

  const newLike = {
    account: 'accounts/123456', // Replace with actual account name
    tweet: 'tweets/789012', // Replace with actual tweet name
  };

  try {
    const likeResponse = await createLike('tweets/789012', newLike);
    if (likeResponse.status === 200) {
      console.log('Like created successfully!');
    } else {
      console.error('Error creating like:', likeResponse.data);
    }
  } catch (error) {
    console.error('Error creating like:', error);
  }

  const newFollowee = {
    followee: 'accounts/789012', // Replace with actual account name
  };

  try {
    const followeeResponse = await createFollowee('accounts/123456', newFollowee);
    if (followeeResponse.status === 200) {
      console.log('Followee created successfully!');
    } else {
      console.error('Error creating followee:', followeeResponse.data);
    }
  } catch (error) {
    console.error('Error creating followee:', error);
  }
})();
